### ISSUE TITLE
 Memory usage: don't render html to memory

### ISSUE THREAD

**Issue by klnusbaum** [klnusbaum/kblog#9]
To_html renders to string. This may be unnecessary since you need html only when you start writing to file. If you make it write to an io::Write, you can defer that until you have a file open.

**Comment by klnusbaum** [comment]
Moved to klnusbaum/kblog#9

**Comment by klnusbaum** [comment]
@abhinav 

I'm currently templating in the html generated by `to_html` . I.e. It's like a two step process:
1. Convert the markdown to html.
2. Then put the resulting html into a template.
(see https://github.com/klnusbaum/kblog/blob/v0.1.0/src/render.rs#L129)

To my knowledge, if I stick with this templating approach, I'm kind of stuck bringing everything into memory, yea?

I could go down a completely different route, where instead of using templates, I stream html fragments into an io::Write.

Wdyt?

**Comment by abhinav** [comment]
Yeah, probably. I haven't dug too deep into templating and memory usage.
TBH, in practice, Markdown files will never be large enough for the memory usage from copying them to ever be a concern.
The original recommendation to play memory golf could be considered a "if you're looking for something to do with the system."
